src/include/migraphx/op/multinomial.hpp

```diff
@@ -21,11 +21,42 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+
+/**
+ * Multinomial or categorical distribution.  Performs a sampling and returns a count of
+ *         each category, or bucket.  This does not require the standard multinomial
+ *         distribution but instead takes a probability distribution as an input.
+ *
+ *          In the large number limit, the fractional counts approach the multinomial distribution.
+ *
+ *      Inputs:   args[0] - a vector of probabilities for each category.  Values are running
+ *                          totals as provided by op prefix_scan_sum.
+ *                          Values are log normalized (i.e. start with any set of numbers > 0, then
+ *                          val[i] = log(val[i]) / sum (log(val[0]) + log(val[1])+ ...) )
+ *                          This input has Rank 2.  Dimension 0 is batch #.  The size of dimension
+ *                          1 is the number of categories.
+ *                args[1] - a vector of random numbers.  Can be 1-, 2- or more dimensions but all
+ *                          except the last dim must be 1.
```

Comment:
I don' understand this comment, the second argument can have dimensions `{batch_size, sample_size}`.

Reply:
I need to add a test with a dynamic batch size > 1 to clarify this.  I'll get back to you.

Reply:
I'm finding problems with the existing logic that takes batch size from arg[0] and another dimension from arg[1].  The only way it would make sense to include a batch dimension in arg[0]--which is the probability distribution--is if we wanted to do multiple multinomials with different probability distributions.  I don't think there's any requirement for such a thing.

Reply:
I refactored the logic and updated the comment.  It's much simpler now.  First argument is a vector, size=number of categories.  Second argument can be any shape, and the output shape is that of the second argument.  There's no special handling for batch size.

Just one bug, under investigation:  it crashes if the second argument has less than 3 dimensions.  Extra size-1 dimensions can be added with Unsqueeze without affecting results.

Reply:
On review, that's what the Onnx spec calls for after all:  you can have multiple CDFs within the same batch.  So, for instance with batch size 2, 5 categories, and 7000 data samples for each distribution, input 1 would be 2x5 and input 2 would be 2x7000.  Input 2 could actually have more dimensions as long as the last dimension is 7000 to deduce the sample_size.  For instance 1x1x2x7000 but I don't think there's any need for that so I changed the comment.